Sources/Crust/Parser.swift

```diff
@@ -493,7 +508,12 @@ extension Parser {
       // either an application or an -> expression. Either way, parse the
       // remaining list of expressions and construct a BasicExprList with the
       // first expression at the beginning.
-      if isStartOfBasicExpr() {
+      if isStartOfBasicExpr() || peek() == .arrow {
+        // If we see an arrow at the start, then consume it and move on.
+        if case .arrow = peek() {
```

Comment:
This isn't safe -- we can't throw away this token because that breaks reprinting from the source.

Reply:
Oh shit, forgot to consume it as a named expression.